### ISSUE TITLE
Failed to upload c++ data by SSL error

### ISSUE THREAD

**Issue by tosyama** [lemurheavy/coveralls-public#1416]
My cpp project failing to upload result from 18 days ago.
I found a SSL error in log of Travis CI.
```
requests.exceptions.SSLError: [Errno 1] _ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure
```
I tried --skip-ssl-verify option to the command 'coveralls', but it was not working.
Is there  any work around?

**Comment by MikeHeiber** [comment]
I am also getting the same apparent bug with my C++ project. After all of the tests finish and it calculates the % of lines covered in each file. I get the traceback:

```Traceback (most recent call last):
  File "/home/travis/.local/bin/coveralls", line 11, in <module>
    sys.exit(run())
  File "/home/travis/.local/lib/python2.7/site-packages/cpp_coveralls/__init__.py", line 106, in run
    return report.post_report(cov_report, args)
  File "/home/travis/.local/lib/python2.7/site-packages/cpp_coveralls/report.py", line 13, in post_report
    verify=(not args.skip_ssl_verify))
  File "/usr/lib/python2.7/dist-packages/requests/api.py", line 88, in post
    return request('post', url, data=data, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/api.py", line 44, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 455, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 558, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/adapters.py", line 385, in send
    raise SSLError(e)
requests.exceptions.SSLError: [Errno 1] _ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure
```
I got the same error skipping SSL verification with
```
coveralls --include src --gcov-options '\-lp' --skip-ssl-verify
```

Anyone know of a workaround for this issue?

**Comment by MikeHeiber** [comment]
This seems to be a Python2 compatibility issue with cpp-coveralls.  I was running Travis CI with the Trusty Ubuntu distribution, using pip to install cpp-coveralls, and then running coveralls.  This used to work just fine.  I guess a newer version of cpp-coveralls must have been released that no longer runs on Python 2.7.  When I updated my Travis CI yaml file to run my coverage test with the Xenial Ubuntu distribution, which defaults to Python 3.6, things work fine again.

**Comment by tosyama** [comment]
Hi @MikeHeiber,
Thank you for your information. I updated travis.yml from trusty to xenial.
It works fine! The coverage of my project updated for the first time in 5 months.

I think the problem come from openssl version or something..  The coveralls-cpp seems to run on python 2.7 still at xenial.

I'll close this issue.
